tests/devices/test_default_qubit.py

```diff
@@ -2243,19 +2243,15 @@ def test_internal_apply_ops_case(self, inverse, monkeypatch):
             res = dev._apply_operation(test_state, op)
             assert np.allclose(res, expected_test_output)
 
-    def test_diagonal_operation_case(self, inverse, mocker, monkeypatch):
```

Comment:
Removed inverse, because `PauliZ().inv()` is old syntax anyways and does not work with the new system. However, the desired `qml.adjoint(PauliZ())` should work, which we do not have to test (we already test somewhere that `adjoint` returns a new op, and diagonal treatment should work for this new op as for any other)!